crates/lang/codegen/src/generator/storage.rs

```diff
@@ -97,6 +97,7 @@ impl Storage<'_> {
                 derive(::ink_storage::traits::StorageLayout)
             )]
             #[derive(::ink_storage::traits::SpreadLayout)]
+            #[derive(::ink_storage::traits::SpreadAllocate)]
```

Comment:
First: You cannot do this since not all `#[ink(storage)]` structs should implement `SpreadAllocate` which is why it is separate from `SpreadLayout` and has not been integrated into the already existing traits.
Second: If this was actually useful (which it isn't) it should reside in another PR.

Reply:
For ink! codegen to automatically implement `SpreadAllocate` optionally we'd need support for Rust trivial trait bounds which currently are unstable.

Reply:
My bad. I thought that the idea behind merging #995 before the `Mapping` was so we could derive `SpreadAllocate` for storage types in the codegen instead of manually deriving it

Reply:
If Rust had trivial trait bounds we actually could do exactly that ... :(